### ISSUE TITLE
Example needed: Using REST API with Macaroons

### ISSUE THREAD

**Issue by Richard87** [lightningnetwork/lnd#730]
How do we use Macaroons with the REST Api?

**Comment by dabura667** [comment]
HTTP header macaroon...?

**Comment by Richard87** [comment]
I finally figured it out, we need to add the macaroon with a `Grpc-Metadata-` prefix in the header, and encode the macaroon file in hex:

```
GET https://localhost:8080/v1/getinfo
Grpc-Metadata-macaroon: hex(admin.macaroon-file)
```

**Comment by brndnmtthws** [comment]
It took me a while to figure this out too. Turns out there's no Java library that supports macaroons v2, and the hex encoding seems a bit non-standard. Anyway, here's what I ended up doing: https://github.com/ompldr/server/blob/master/src/main/kotlin/org/ompldr/server/models/Invoice.kt#L73-L107

tl;dr: read in the admin macaroon as a byte array, encode it as hex, et voila.

**Comment by Roasbeef** [comment]
@brndnmtthws how about making a PR adding some docs w.r.t to how to do that generically? (so on the HTTP level).

**Comment by Roasbeef** [comment]
Ah somehow missed @Richard87's example above. But either way a PR to update the docs would be appreciated, would be merged swiftly.

**Comment by brndnmtthws** [comment]
Have a look at #785. I added a code sample, but if that's too much I can yank it. Or perhaps there should be a separate file that contains code samples in various languages?

**Comment by Richard87** [comment]
If we should add a samples, I think it should be as simple as possible :)

For example just a curl script that lets the users experiment with it and see what happens:

```
curl --insecure --header "Grpc-Metadata-macaroon: $(xxd -ps -u -c 1000  ./var/lnd/admin.macaroon)" https://localhost:8080/v1/getinfo
```

**Comment by brndnmtthws** [comment]
Hmm, okay, I'll add that to the doc.

However, in my opinion at least, providing proper examples with 'real code' will make it significantly easier for people to get started with lnd.

**Comment by Richard87** [comment]
Yup, that I completley agree with, just Java is so incredibly verbose, that it's hard to see the core of what is needed...

for example Node.js:
```js
var macaroonHexString = Buffer.from(fs.readFileSync(adminMacaroonFile), 'utf8').toString('hex');

request.get({
  url: 'https://localhost:8080/v1/getinfo',
  rejectUnauthorized: false,
  headers: {
    'Grpc-Metadata-macaroon': macaroonHexString
  }
});
```

**Comment by Roasbeef** [comment]
Fixed by #785.
